### ISSUE TITLE
Webpack Trouble

### ISSUE THREAD

**Issue by Workruft** [ryanking1809/THREE.MeshLine#14]
No wonder what I do, I can't get this or the original library to work with Webpack. I've tried `require` and `import` both.

It seems I can get it to "load" but then it's not drawing things like it was when I was importing it with `<script>`. Maybe the `export`ing process can be updated?

**Comment by ryanking1809** [comment]
Sounds odd, I have no troubles with webpack. Are you getting any sort of errors? Or do you have anything that can reproduce the issue?

**Comment by AndrewDavis** [comment]
Orly? Is there some code of yours online that I can look at then that has it working?

**Comment by Workruft** [comment]
Orly? Is there some code of yours online that I can look at then that has it working?

**Comment by ryanking1809** [comment]
This should work fine  https://codesandbox.io/s/canvas-sketch-a4-artwork-xlsto

**Comment by Workruft** [comment]
No, I mean with Webpack. This is just using ES6 `import`.

**Comment by ryanking1809** [comment]
I don't understand, don't you just write the code like above then bundle it with webpack? The import syntax shouldn't differ.

Personally, I just use `create-react-app` and it uses webpack under the hood. So assume it's
something to do with your code or webpack config?
.

**Comment by ryanking1809** [comment]
Oh looks like you're using webpack plugins. I searched your repo, and it looks like you don't have a MeshLineMaterial anywhere. If it's up to date, I imagine that's the issue.

**Comment by ryanking1809** [comment]
Oh, wait I found it. You're not getting any errors? If you console.log MeshLine, what do you get?

**Comment by Workruft** [comment]
> I don't understand, don't you just write the code like above then bundle it with webpack? The import syntax shouldn't differ.

Webpack does something with all the imports by shifting it all around and making it not use ES6 modules and using functions and parameters and stuff.

I do not get errors (at least not with the way I'm currently doing it). It "loads", but once I started using the node package and Webpack, everything else works with the same code except for the MeshLine stuff. I see an object with the 3 properties in the console. It's just not drawing my circles.

**Comment by Workruft** [comment]
`THREE` should be accessible as a global when that code is run. I suspect it all has something to do with the funky way MeshLine's exporting system works, or maybe the `root` or `this` stuff, or maybe something else idk.

I don't know enough about Webpack or MeshLine or imports to say any more.

**Comment by Workruft** [comment]
For who knows what reason, I can't use `import { MeshLine } ...` as you do. It breaks my use of `module.exports =`, which I have to use to get the rest of the code to work (it doesn't like my use of `export` and `import` I suppose because of Webpack).

So I use `let MeshLine = require('threejs-meshline');` and then in the console that gives me:
```
{…}
MeshLine: function MeshLine()​
MeshLineMaterial: function MeshLineMaterial(parameters)​
MeshLineRaycast: function MeshLineRaycast(raycaster, intersects)
```

But if I use it like `new MeshLine.MeshLine()` and `MeshLine.MeshLineMaterial` etc., it doesn't draw.

See my last 2 commits for more info.

**Comment by ryanking1809** [comment]
Hmm ok not sure what's going on there, could probably take a look this weekend. Not 100% sure I know enough about webpack to help but let's see.

A really nice bundler you also might like to try is called `parcel`, it has a really simple setup and is much nice to use than webpack I find.
https://parceljs.org/getting_started.html

**Comment by Workruft** [comment]
> A really nice bundler you also might like to try is called `parcel`, it has a really simple setup and is much nice to use than webpack I find.
> https://parceljs.org/getting_started.html

Huh, that does look nice! I'll check it out, thanks.

**Comment by Workruft** [comment]
_**Beautiful!**_

https://github.com/Workruft/Workruft/commit/a04c74a3d7dbfb35ef977a4fc4918009209761cf
https://github.com/Workruft/Workruft/commit/82957077bebce64b70124c3c9c0b07ee8f91d262

ParcelJS is working excellently, thank you very much!

I can even use ThreeJS, MeshLine, and my own code as globals if I want. Sooooo much more intuitive for it to just automatically package everything from a starting HTML file.

See my second URL; I did have to disable [HMR](https://parceljs.org/hmr.html) (don't care for it anyways), and to make it work without serving it up I had to slightly alter the URL it uses. But now it's doing exactly what I want.

Goodbye Webpack...
